.github/workflows/release.yml

```diff
@@ -0,0 +1,47 @@
+name: Publish chart upon release creation
+
+on:
+  release:
+    types:
+      - "created"
+
+jobs:
+  release-chart:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            fetch-depth: 0
+            submodules: recursive
+
+      - uses: cachix/install-nix-action@v22
+      - name: Pre-populate nix-shell
+        run: |
+          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+          nix-shell --pure --run "echo" ./shell.nix
+
+      - name: Remove prerelease label from dependencies
+        run: |
+            yq -i '(.dependencies[] | select(.name == "localpv-provisioner") | .version) |= sub("-prerelease$", "")' charts/Chart.yaml
+            yq -i '(.dependencies[] | select(.name == "zfs-localpv") | .version) |= sub("-prerelease$", "")' charts/Chart.yaml
+            yq -i '(.dependencies[] | select(.name == "lvm-localpv") | .version) |= sub("-prerelease$", "")' charts/Chart.yaml
+            yq -i '(.dependencies[] | select(.name == "mayastor") | .version) |= sub("-prerelease$", "")' charts/Chart.yaml
+
+      - name: Check if the chart is publishable
+        run: |
+          TAG=${{ github.event.release.tag_name }}
+          nix-shell --pure --run "./scripts/helm/update-chart-version.sh --tag $TAG --publish-release" ./shell.nix
```

Comment:
this is not enough, since we've updated the versions we have to re-generate the Chart.yaml metadata

Reply:
Ah, I was under the impression that the chart-images.yaml (Helm Chart Images CI) would do that. Should I extend the workflow to include 
on:
  release:
    types:
      - "created"
OR 
Do the steps explicitly in this workflow.

Reply:
I'm confused, how is extending the chart-image helping here?

Reply:
I have extended chart-images.yml to be used as a resubale workflow and calling it from branch preparation to avoid dupilcation of code.

Reply:
But this won't work, the contents on this job won't get updated

Reply:
Added the steps to each workflow.